bot/exts/utils/reminders.py

```diff
@@ -37,11 +38,43 @@
 LOCK_NAMESPACE = "reminder"
 WHITELISTED_CHANNELS = Guild.reminder_whitelist
 MAXIMUM_REMINDERS = 5
+REMINDER_EDIT_CONFIRMATION_TIMEOUT = 60
 
 Mentionable = discord.Member | discord.Role
 ReminderMention = UnambiguousUser | discord.Role
 
 
+class ModifyReminderConfirmationView(discord.ui.View):
+    """A view to confirm modifying someone else's reminder by admins."""
+
+    def __init__(self, author: discord.Member):
+        super().__init__(timeout=REMINDER_EDIT_CONFIRMATION_TIMEOUT)
+        self.author = author
+        self.result: bool | None = None
+
+    async def interaction_check(self, interaction: Interaction) -> bool:
+        """Only allow interactions from the command invoker."""
+        return interaction.user.id == self.author.id
+
+    async def on_timeout(self) -> None:
+        """Default to not modifying if the user doesn't respond."""
+        self.result = False
```

Comment:
If the view times out, the view doesn't get deleted so the buttons fail with "application didn't respond."
Should we manually delete the view if it timed out (with the return value of `View.wait()`), or should all deleting (regardless of whether it timed out) happen in `_can_modify`?

Reply:
We recently had a discussion about deleting messages like this. And it's not something we want to keep on doing.
Let's keep the view, but delete the buttons.